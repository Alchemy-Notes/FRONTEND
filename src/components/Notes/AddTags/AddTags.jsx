import React, { useState } from 'react';
import Button from '../../Button/Button';

function AddTags({ tags, setTags }) {
  const [tag, setTag] = useState('');
  const handleSubmit = (e) => {
    e.preventDefault();
    setTags((prevState) => [...prevState, tag]);
    setTag('');
  };

  return (
    <section>
      {/* make a card for each tag with an x to delete a tag from a note */}
      <div>{tags.map((tag) => tag + ' ')}</div>
      <form onSubmit={handleSubmit}>
        <label htmlFor="tags">Tags </label>
        <input
          id="tags"
          value={tag}
          type="textbox"
          onChange={(e) => {
            setTag(e.target.value);
          }}
        />
        <Button type={'submit'} buttonText="Add Tag" />
      </form>
    </section>
  );
}

export default AddTags;
