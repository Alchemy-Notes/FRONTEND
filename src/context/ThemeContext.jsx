import { createContext, useContext, useState } from 'react';

const ThemeContext = createContext();

const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState(false);

  return (
    <ThemeContext.Provider value={{ theme, setTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

const useTheme = () => {
  const context = useContext(ThemeContext);

  if (context === undefined) {
    throw new Error('useTheme must be used within a ThemeContext Provider');
  }

  return context;
};

export { ThemeProvider, useTheme };
