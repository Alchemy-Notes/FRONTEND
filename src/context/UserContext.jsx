import { createContext, useContext, useEffect, useState } from 'react';
import { getMe } from '../services/auth';
import { getUserNotes } from '../services/notes';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const [notes, setNotes] = useState(null);
  const [tree, setTree] = useState(null);

  useEffect(() => {
    async function get() {
      const result = await getMe();
      setUser(result);
    }
    get();
  }, []);

  useEffect(() => {
    async function getNotes() {
      const result = await getUserNotes({
        userId: user.id,
        query: { type: 'favorite' },
      });
      setNotes(result);
    }
    if (user && user.id) getNotes();
  }, [user]);

  return user && notes ? (
    <UserContext.Provider
      value={{ user, setUser, notes, setNotes, tree, setTree }}
    >
      {children}
    </UserContext.Provider>
  ) : (
    <></>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserContext Provider');
  }

  return context;
};

export { UserProvider, useUser };
