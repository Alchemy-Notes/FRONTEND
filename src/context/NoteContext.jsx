import { createContext, useContext, useEffect, useState } from 'react';
import { getUserNotes } from '../services/notes';
import { useUser } from './UserContext';

const NoteContext = createContext();

const NoteProvider = ({ children }) => {
  const [notes, setNotes] = useState([]);
  const { user } = useUser();

  useEffect(() => {
    async function get() {
      const result = await getUserNotes({
        userId: user.id,
        query: { type: 'favorite' },
      });
      setNotes(result);
    }
    if (user && user.id) get();
  }, []);

  return (
    <NoteContext.Provider value={{ notes, setNotes }}>
      {children}
    </NoteContext.Provider>
  );
};

const useNotes = () => {
  const context = useContext(NoteContext);

  if (context === undefined) {
    throw new Error('useNotes must be used within a NoteContext Provider');
  }

  return context;
};

export { NoteProvider, useNotes };
