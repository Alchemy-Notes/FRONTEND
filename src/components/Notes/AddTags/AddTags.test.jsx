import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom/';
import { UserProvider } from '../../../context/UserContext';
import AddTags from './AddTags';

it('should render our button to log into github', () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider>
        <AddTags />
      </UserProvider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
