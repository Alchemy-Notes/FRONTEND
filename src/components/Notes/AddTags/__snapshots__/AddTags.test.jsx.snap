// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render our button to log into github 1`] = `
<div>
  <section>
    <div>
       
    </div>
    <form>
      <label
        for="tags"
      >
        Tags 
      </label>
      <input
        id="tags"
        value=""
      />
      <button
        type="submit"
      >
        Add Tag
      </button>
    </form>
  </section>
</div>
`;
